src/lib/operators/join_sort_merge/radix_cluster_sort.hpp

```diff
@@ -66,6 +68,26 @@ class RadixClusterSort {
 
   virtual ~RadixClusterSort() = default;
 
+  template <typename T2>
+  static std::enable_if_t<std::is_integral_v<T2>, size_t> get_radix(T2 value, size_t radix_bitmask) {
+    return static_cast<int64_t>(value) & radix_bitmask;
+  }
+
+  template <typename T2>
+  static std::enable_if_t<std::is_floating_point_v<T2>, size_t> get_radix(T2 value, size_t radix_bitmask) {
+    PerformanceWarning("Using hash to perform bit_cast/radix partitioning of floating point number");
+    return std::hash<T2>{}(value)&radix_bitmask;
+  }
+
+  // Radix calculation for non-arithmetic types
+  template <typename T2>
+  static std::enable_if_t<std::is_same_v<T2, pmr_string>, uint32_t> get_radix(T2 value, size_t radix_bitmask) {
+    PerformanceWarning("Using memcpy to perform bit_cast/radix partitioning of strings");
+    uint32_t radix;
+    std::memcpy(&radix, value.c_str(), std::min(value.size(), sizeof(radix)));
```

Comment:
Reason to not use std::hash here?

Reply:
strings used memcpy before. But this is actually UB, as the memcpy might access memory it shouldn't for strings <= 3 chars. will use std::hash then.